### ISSUE TITLE
phone-ru.js

### ISSUE THREAD

**Issue by Soul-git** [RobinHerbots/jquery.inputmask#1158]
Hello! When connecting the phone-ru.js work masks are not stable , the field is loaded and processed for a long time . Can you solve this problem ?

**Comment by Soul-git** [comment]
https://jsfiddle.net/y2d07raL/

**Comment by RobinHerbots** [comment]
@Soul-git , 

I know It takes too long.  

The russian phone codes contain alot of entries.  To make it really usefull I should rewrite the phone extension.  The idea would be to initially load only the generic regions and while entering data load the more specific region.  This will reduce the number of variations in the mask which means more speed.  

But the question is when do I have some spare-time...

**Comment by Soul-git** [comment]
Yes, I know that you know about the problem . I wanted to ask , at what stage of the solution of this issue . Thank you. (sorry , I write through an online translator ).

**Comment by andr-04** [comment]
I've adapted my plugin to latest inputmask. See [the demo] (http://andr-04.github.io/inputmask-multi/ru.html).

**Comment by Soul-git** [comment]
Thank you. But unfortunately , with the appearance of your plangina before its renovation took place 2 years . jquery.inputmask constantly updated , I want to be sure that your plangin will be maintained up to date , and will not cease to work at the next Major Upgrade jquery.inputmask.

**Comment by andr-04** [comment]
@Soul-git, yes, I had no plans to adopt my plugin for inputmask v3.x, but recently I've got the [pull request] (https://github.com/andr-04/inputmask-multi/pull/18). It indicates that inputmask-multi plugin is needed so I decided to merge this and fix bugs after that. So if you like inputmask-multi you can always give pull requests -- it will help to maintain it to work with next Major Upgrades of jquery.inputmask.

**Comment by Soul-git** [comment]
Not very convenient to communicate in English . I wrote on a page of your plangin .

**Comment by ichepurnoy** [comment]
@andr-04 Thanks for your work, Andrey! Can you please update your webpage to indicate the version/release date, because people would probably miss the fact that you've updated your plugin. We've recently had to remove its old version, due to the upgrade of Inputmask. Doh, we've just finished, and here comes the update!)

**Comment by RobinHerbots** [comment]
@Soul-git , 

Ok, got some performance tweaks in a local branch, which can handle the many russian phone codes., ... but it has still some side effects I have to solve....  but it will come soon now

**Comment by RobinHerbots** [comment]
Updated fiddle https://jsfiddle.net/y2d07raL/4/

**Comment by ichepurnoy** [comment]
dear Robin, what exactly does this commit solve/enhance ?
BTW, I see that deleting the last digit from the region code still takes ~ a lot of time and CPU.

**Comment by RobinHerbots** [comment]
@ichepurnoy , 

Yes that I still need to investigate.  Entry performance should be alot better.  Also the caret placing is better.

Todo is the delete performance (in some cases) and define the correct placeholder according the entry (also for some cases)

**Comment by ichepurnoy** [comment]
@RobinHerbots , the demo of Andrey @andr-04 works fine, as it seems. [http://andr-04.github.io/inputmask-multi/ru.html](url)

I was able to immediately add and delete digits. 
Try  the code +7(42352) , and compare your demo and his. Your demo, when deleting digits, deletes the hole code at once, when you reach it by Backspace. His demo deletes digits 1 by 1, I don't know , maybe you too have some setting to set it like this. Maybe you can try his solution, if he does not mind.

**Comment by RobinHerbots** [comment]
@ichepurnoy , 

I know the solution of @andr-04 works perfect, but for me it is about the alternation logic and not about a specific solution.  Working on this, and take the russian phone codes as test case, improves the generic alternation logic and the inputmask as a whole.

**Comment by Soul-git** [comment]
Thank you for working in this direction . Speed ​​is already a lot better.
Work @ andr- 04 is good, but can not keep up with your updates . Besides not work when compressive html.

**Comment by Tuning-Soft** [comment]
Hi! The second year waiting for this feature, really looking forward to, make a donation today, thank you!!!

**Comment by RobinHerbots** [comment]
@ichepurnoy ,  @andr-04 

Have you ever tried to enter the following in the inputmask-multi

"+7(341)###-##-##",  //ok 
"+7(3412)##-##-##",  //fails
"+7(34126)#-##-##",  //fails
"+7(34130)#-##-##",  //fails

In my opinion the generic mask ("+7(341)###-##-##",) need to be removed from the phonecodes.  I can give you also other examples where it doesn't work.

**Comment by RobinHerbots** [comment]
@Tuning-Soft , 

With the current on github there are 14 phones failing of the 2425

**Comment by RobinHerbots** [comment]
Performance update issue => see #750

**Comment by RobinHerbots** [comment]
@Soul-git ,  @Tuning-Soft , @ichepurnoy , @andr-04 ,

The performance of entry should be good now.  Can you have a try with the current version on github

Todo:
I need to refactor the getmetadata to retrieve the region etc and I need some tweaking for deletes.

**Comment by RobinHerbots** [comment]
@Tuning-Soft , 

Getmetadata also fixed.

Can you have a try.
